Samples/BackgroundBlur/BackgroundBlur/SegmentModel.cpp

```diff
@@ -0,0 +1,393 @@
+
+		// v3: get from a d3dresource
+		/*ID3D12Resource* res = NULL;
+		HRESULT hr = foreground.as<ITensorNative>()->GetD3D12Resource(&res);
+		UINT DstRowPitch = 0, DstDepthPitch = 0, SrcSubresource = 0;
+		hr = res->ReadFromSubresource((void*)*pDest, DstRowPitch, DstDepthPitch, SrcSubresource, NULL);*/
+		return;
+	}
+	else 
+	{
+		auto reference = foreground.as<TensorUInt8Bit>().CreateReference().data();
+		CopyMemory(*pDest, reference, cbImageSize);
+	}
+}
+
+void SegmentModel::RunTest(const BYTE** pSrc, BYTE** pDest, DWORD cbImageSize) 
+{
+	// Right now the single input type I allow is topdown, this should work
+	winrt::array_view<const byte> source{*pSrc, *pSrc + cbImageSize}; // TODO: Does this work when topdown vs. bottomup
+	std::vector<int64_t> shape = { 1, m_imageHeightInPixels, m_imageWidthInPixels, 4 };
+
+	ITensor inputRawTensor = TensorUInt8Bit::CreateFromArray(std::vector<int64_t>{1, cbImageSize}, source);
+	ITensor outputTensor = TensorUInt8Bit::Create(shape);
+	auto binding = Evaluate(m_sess, { &inputRawTensor }, &outputTensor, true);
+
+	UINT32 outCapacity = 0;
+	if (m_useGPU)
+	{
+		// v1: just get the reference- should fail
+		auto reference = outputTensor.as<TensorUInt8Bit>().CreateReference().data();
+		CopyMemory(*pDest, reference, cbImageSize);
+
+		// v3: get from a d3dresource
+		/*ID3D12Resource* res = NULL;
+		HRESULT hr = outputTensor.as<ITensorNative>()->GetD3D12Resource(&res);
+		UINT DstRowPitch = 0, DstDepthPitch = 0, SrcSubresource = 0;
+		hr = res->ReadFromSubresource((void*)*pDest, DstRowPitch, DstDepthPitch, SrcSubresource, NULL);*/
+		return;
+	}
+}
+
+void SegmentModel::SetImageSize(UINT32 w, UINT32 h) 
+{
+	m_imageWidthInPixels = w;
+	m_imageHeightInPixels = h;
+}
+
+LearningModel SegmentModel::Argmax(long axis, long h, long w)
+{
+	auto builder = LearningModelBuilder::Create(12)
+		.Inputs().Add(LearningModelBuilder::CreateTensorFeatureDescriptor(L"Data", TensorKind::Float, array_view<int64_t const>{ -1, -1, h, w })) // Different input type? 
+		.Outputs().Add(LearningModelBuilder::CreateTensorFeatureDescriptor(L"Output", TensorKind::Float, array_view<int64_t const>{ -1, -1, h, w })) // Output of int64? 
+		.Operators().Add( LearningModelOperator(L"ArgMax")
+			.SetInput(L"data", L"Data")
+			.SetAttribute(L"keepdims", TensorInt64Bit::CreateFromArray(std::vector<int64_t>{1}, std::vector<int64_t>{ 1 }))
+			.SetAttribute(L"axis", TensorInt64Bit::CreateFromIterable(std::vector<int64_t>{1}, std::vector<int64_t> { axis })) // Correct way of passing axis? 
+			.SetOutput(L"reduced", L"Reduced"))
+		.Operators().Add( LearningModelOperator(L"Cast")
+			.SetInput(L"input", L"Reduced")
+			.SetAttribute(L"to", TensorInt64Bit::CreateFromIterable(std::vector<int64_t>{}, std::vector<int64_t>{OnnxDataType::ONNX_FLOAT}))
+			.SetOutput(L"output", L"Output"))
+		;
+
+	return builder.CreateModel();
+}
+
+LearningModel SegmentModel::FCNResnet()
+{
+	auto rel = std::filesystem::current_path();
```

Comment:
std::filesystem is C++20

Reply:
Looks like it's c++17 as well: https://en.cppreference.com/w/cpp/filesystem/current_path

Reply:
Oh I was sure it didn't make it into C++17. You're correct, my bad.